rosidl_generator_rs/resource/action.rs.em

```diff
@@ -0,0 +1,205 @@
+    pre_field_serde=pre_field_serde,
+    get_idiomatic_rs_type=get_idiomatic_rs_type,
+    constant_value_to_rs=constant_value_to_rs)
+
+TEMPLATE(
+    'srv_rmw.rs.em',
+    package_name=package_name, interface_path=interface_path,
+    srv_specs=action_srv_specs,
+    get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
+    get_idiomatic_rs_type=get_idiomatic_rs_type,
+    constant_value_to_rs=constant_value_to_rs)
+}@
+}  // mod rmw
+
+@{
+TEMPLATE(
+    'msg_idiomatic.rs.em',
+    package_name=package_name, interface_path=interface_path,
+    msg_specs=action_msg_specs,
+    get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
+    get_idiomatic_rs_type=get_idiomatic_rs_type,
+    constant_value_to_rs=constant_value_to_rs)
+}@
+
+@{
+TEMPLATE(
+    'srv_idiomatic.rs.em',
+    package_name=package_name, interface_path=interface_path,
+    srv_specs=action_srv_specs,
+    get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
+    get_idiomatic_rs_type=get_idiomatic_rs_type,
+    constant_value_to_rs=constant_value_to_rs)
+}@
+
+@[for subfolder, action_spec in action_specs]
+
+@{
+type_name = action_spec.namespaced_type.name
+}@
+
+#[link(name = "@(package_name)__rosidl_typesupport_c")]
+extern "C" {
+    fn rosidl_typesupport_c__get_action_type_support_handle__@(package_name)__@(subfolder)__@(type_name)() -> *const std::os::raw::c_void;
+}
+
+// Corresponds to @(package_name)__@(subfolder)__@(type_name)
+pub struct @(type_name);
+
+impl rosidl_runtime_rs::Action for @(type_name) {
+  type Goal = crate::@(subfolder)::@(type_name)@(ACTION_GOAL_SUFFIX);
+  type Result = crate::@(subfolder)::@(type_name)@(ACTION_RESULT_SUFFIX);
+  type Feedback = crate::@(subfolder)::@(type_name)@(ACTION_FEEDBACK_SUFFIX);
+
+  fn get_type_support() -> *const std::os::raw::c_void {
+    // SAFETY: No preconditions for this function.
+    unsafe { rosidl_typesupport_c__get_action_type_support_handle__@(package_name)__@(subfolder)__@(type_name)() }
+  }
+}
+
+impl rosidl_runtime_rs::ActionImpl for @(type_name) {
+  type GoalStatusMessage = action_msgs::msg::rmw::GoalStatusArray;
+  type FeedbackMessage = crate::@(subfolder)::rmw::@(type_name)@(ACTION_FEEDBACK_MESSAGE_SUFFIX);
+
+  type SendGoalService = crate::@(subfolder)::rmw::@(type_name)@(ACTION_GOAL_SERVICE_SUFFIX);
+  type CancelGoalService = action_msgs::srv::rmw::CancelGoal;
+  type GetResultService = crate::@(subfolder)::rmw::@(type_name)@(ACTION_RESULT_SERVICE_SUFFIX);
+
+  fn create_goal_request(
+    goal_id: &[u8; 16],
+    goal: crate::@(subfolder)::rmw::@(type_name)@(ACTION_GOAL_SUFFIX),
+  ) -> crate::@(subfolder)::rmw::@(type_name)@(ACTION_GOAL_SERVICE_SUFFIX)@(SERVICE_REQUEST_MESSAGE_SUFFIX) {
+    crate::@(subfolder)::rmw::@(type_name)@(ACTION_GOAL_SERVICE_SUFFIX)@(SERVICE_REQUEST_MESSAGE_SUFFIX) {
+      goal_id: unique_identifier_msgs::msg::rmw::UUID { uuid: *goal_id },
```

Comment:
I thought we didn't want to have a dependency on `unique_identifier_msgs` here

Reply:
We want to avoid creating a dependency on `unique_identifier_msgs` from the `rosidl_runtime_rs` package, since the latter is a dependency of all message packages. This would cause a cyclic dependency.

However, the crates generated by the `rosidl_generator_rs` do have dependencies on any message packages they use as fields. In this case, `unique_identifier_msgs/msg/UUID` is a field of the underlying goal service request type.

Reply:
Ahh, I see. So the function signature of `create_goal_request` can't reference `unique_identifier_msgs` but the impl of that function actually can. 

Yeah, we should explore some of the ideas discussed earlier to simplify this. Not needed for this PR though.